Q1. Pearson correlation coefficient is a measure of the linear relationship between two variables. Suppose
you have collected data on the amount of time students spend studying for an exam and their final exam
scores. Calculate the Pearson correlation coefficient between these two variables and interpret the result.


In [8]:
import pandas as pd
data = {
    'Study_Time': [3, 5, 2, 4, 6, 1, 7, 2, 4, 5],
    'Exam_Scores': [70, 85, 60, 75, 90, 50, 95, 65, 80, 85]
}

df = pd.DataFrame(data)
# Pearson Correlation Coefficient
p_c_coff = df['Study_Time'].corr( df['Exam_Scores'])
print("Pearson Correlation Coefficient:",p_c_coff)


Pearson Correlation Coefficient: 0.98216253380743



Interpret the Result

Since the calculated Pearson correlation coefficient is  0.98216253380743 , it would indicate a strong positive linear relationship between the amount of time students spend studying for the exam and their final exam scores. In other words, as the study time increases, the exam scores tend to increase as well.

Q2. Spearman's rank correlation is a measure of the monotonic relationship between two variables.
Suppose you have collected data on the amount of sleep individuals get each night and their overall job
satisfaction level on a scale of 1 to 10. Calculate the Spearman's rank correlation between these two
variables and interpret the result.

In [18]:
import pandas as pd
data1 = {
    'sleep_time': [7, 5, 8, 9, 4, 8, 7, 6],
    'job_satisfaction': [8, 9, 8, 5, 4, 7, 6, 5]
}

df1 = pd.DataFrame(data1)
df_r = df1.corr(method = "spearman")

corr = df_r['sleep_time']['job_satisfaction']

# Interpretation of the result
if corr > 0:
    intr = "There is a positive monotonic relationship between sleep and job satisfaction."
elif corr < 0:
    intr = "There is a negative monotonic relationship between sleep and job satisfaction."
else:
    intr = "There is no monotonic relationship between sleep and job satisfaction."

print("Spearman's Rank Correlation:", corr)
print("Interpretation:", intr)


Spearman's Rank Correlation: 0.10975609756097561
Interpretation: There is a positive monotonic relationship between sleep and job satisfaction.


Q3. Suppose you are conducting a study to examine the relationship between the number of hours of
exercise per week and body mass index (BMI) in a sample of adults. You collected data on both variables
for 50 participants. Calculate the Pearson correlation coefficient and the Spearman's rank correlation
between these two variables and compare the results.

In [19]:
import numpy as np

num_participants = 50
hours_of_exercise = np.random.uniform(1, 10, size=num_participants)
bmi = np.random.uniform(18, 35, size=num_participants)
# print(bmi,'\n',hours_of_exercise )

In [20]:
print("below is the sample data\n hours_of_exercise\n",hours_of_exercise,'\n\n bmi\n',bmi)

below is the sample data
 hours_of_exercise
 [2.55581694 3.48539532 4.86573223 7.94539366 2.95310634 7.0275219
 4.48090737 7.5765219  7.81687715 7.85306521 9.342681   4.25372474
 7.06474685 1.53996691 3.31831312 5.2390998  9.64516874 3.29605898
 6.17725357 3.43036173 3.73670545 8.58352887 6.68386327 8.37936146
 1.27855218 7.0718062  1.2074493  5.28057697 7.38146007 5.67666941
 9.70316896 9.92508444 1.67001908 1.3107684  7.9588129  3.39482414
 6.34742867 7.99249637 2.16784256 6.17519534 2.28199732 7.30385236
 7.45702751 5.73905035 9.32668789 8.17551296 5.9634002  6.14534453
 8.43273986 2.1602956 ] 

 bmi
 [28.23052223 20.60743725 22.92313651 31.63791323 30.52330054 25.04215245
 20.8089926  33.89401357 24.18877308 19.16731392 32.59492483 26.9483361
 18.59183194 33.36928961 22.4598744  24.04205186 33.48330688 25.79649548
 27.30511002 29.22229217 33.85733623 29.36398404 23.36696513 18.0479196
 28.85600737 31.83476203 21.14606315 26.10811189 20.42932935 27.7844015
 21.00025869 24.74257817 2

In [23]:
import pandas as pd
data3 = {
    'hours_of_exercise': hours_of_exercise,
    'bmi': bmi
}
df3 = pd.DataFrame(data3)

df_pear = df3.corr(method='pearson')
df_spear = df3.corr(method='spearman')

corr_pear = df_pear['hours_of_exercise']['bmi']
corr_spear = df_spear['hours_of_exercise']['bmi']


comp = ""
if abs(corr_pear) > abs(corr_spear):
    comp = "The Pearson correlation coefficient is larger in magnitude."
elif abs(corr_pear) < abs(corr_spear):
    comp = "The Spearman's rank correlation is larger in magnitude."
else:
    comp = "Both correlation coefficients are of the same magnitude."

print("Pearson Correlation Coefficient:", corr_pear)
print("Spearman's Rank Correlation:", corr_pear)
print("Comparison:", comp)

Pearson Correlation Coefficient: 0.08478678343615757
Spearman's Rank Correlation: 0.08478678343615757
Comparison: The Spearman's rank correlation is larger in magnitude.


Q4. A researcher is interested in examining the relationship between the number of hours individuals
spend watching television per day and their level of physical activity. The researcher collected data on
both variables from a sample of 50 participants. Calculate the Pearson correlation coefficient between
these two variables.

In [24]:
import numpy as np
import pandas as pd

num_participants = 50
hours_watching_tv = np.random.uniform(0, 6, size=num_participants)  
physical_activity = np.random.uniform(1, 10, size=num_participants)  


In [25]:
print("below is the sample data\n hours_watching_tv\n",hours_watching_tv,'\n\n physical_activity\n',physical_activity)

below is the sample data
 hours_watching_tv
 [1.96591548 5.65324074 3.08487858 3.68903339 1.10614193 3.78458866
 0.54893053 3.31252047 3.58945084 2.27113446 1.11600648 3.93187217
 1.5303877  1.61500507 4.8707411  4.55675424 0.97088377 3.6065057
 2.09538803 1.42858551 3.73381365 5.90823511 5.45466132 1.72000453
 0.3155427  1.58051944 3.27582536 3.7271097  2.95377359 2.6534022
 3.5180453  2.74614726 1.74465245 4.81226575 1.16046815 4.49678461
 4.24403984 3.81949923 2.58288402 4.51334043 4.23236223 3.27076129
 3.4375454  5.41931292 3.43751973 2.43739222 5.46703505 4.98930623
 5.9246741  4.48882767] 

 physical_activity
 [3.66119295 1.078349   3.33200154 6.72057989 6.03304268 7.12366256
 7.63569687 8.21325508 3.99753679 7.41197994 2.6975692  6.35518834
 5.84133241 8.1102802  9.36249156 8.92958498 1.1556266  9.95544933
 4.49226005 1.48046135 4.45005892 3.42200253 7.36899339 5.56171965
 1.98483115 7.45974029 6.97797309 1.19220581 8.76928487 5.89195145
 4.91071333 7.12661422 8.42519942 4.9999

In [27]:
data4 = pd.DataFrame({
    'Hours_Watching_TV': hours_watching_tv,
    'Physical_Activity': physical_activity
})

# Pearson correlation coefficient
corr_4 = data4['Hours_Watching_TV'].corr(data4['Physical_Activity'])

print("Pearson Correlation Coefficient:", corr_4)

Pearson Correlation Coefficient: 0.15317037959924196


Q5. A survey was conducted to examine the relationship between age and preference for a particular
brand of soft drink. The survey results are shown below:
Age(Years) softdrink preference
25 Coke
42 Pepsi
37 Mountain dew
19 Coke
31 Pepsi
28 Coke

In [32]:
survey_data = {
    'Age': [25, 42, 37, 19, 31, 28],
    'SoftDrink_Preference': ['Coke', 'Pepsi', 'Mountain Dew', 'Coke', 'Pepsi', 'Coke']
}

df = pd.DataFrame(survey_data)

#label encoding
label_mapping = {'Coke': 0, 'Pepsi': 1, 'Mountain Dew': 2}
df['SoftDrink_Preference_Num'] = df['SoftDrink_Preference'].map(label_mapping)

#Pearson correlation coefficient
corr_5 = df['Age'].corr(df['SoftDrink_Preference_Num'])
print("Pearson Correlation Coefficient between Age and SoftDrink Preference:", corr_5)


Pearson Correlation Coefficient between Age and SoftDrink Preference: 0.7587035441865058


Q6. A company is interested in examining the relationship between the number of sales calls made per day
and the number of sales made per week. The company collected data on both variables from a sample of
30 sales representatives. Calculate the Pearson correlation coefficient between these two variables.


In [36]:
import numpy as np
import pandas as pd

size = 30
calls = np.random.randint(7, 25, size=size)  # calls per day btwn 7-25
sales = np.random.randint(2, 15, size=size)  # sales per week btwn 2-15
print('calss: ',calls,'\n\nsales: ',sales)

calss:  [23  9 12 23 11  7 17 24 14 12 13 14 11 17 16 10 11 12 15 18 18 10  9 11
 14 16 14  9 11 14] 

sales:  [ 6  4  3  2  4 14  5 10 12  2  6 11  5  9 14  4  2 11  2 10  6  6 11  5
  3 14  3  3  8  5]


In [39]:
df6 = pd.DataFrame({
    'Sales_Calls_Per_Day': calls,
    'Sales_Per_Week': sales
})

# Pearson correlation coefficient
corr_6 = df6['Sales_Calls_Per_Day'].corr(df6['Sales_Per_Week'])

print("Pearson Correlation Coefficient between sales calls made per day and the number of sales made per week is:\n", corr_6)


Pearson Correlation Coefficient between sales calls made per day and the number of sales made per week is:
 0.06386170197256727
